core/domain/exp_services.py

```diff
@@ -1982,6 +1983,36 @@ def create_or_update_draft(
     exp_user_data.put()
 
 
+def try_upgrading_draft_to_exp_version(exp_id, user_id):
```

Comment:
Please add tests for all new code in this file.

Reply:
Added unit test in draft services and exp services